onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -279,14 +279,20 @@ common::Status TensorrtExecutionProvider::Compile(const std::vector<onnxruntime:
     // Create TensorRT engine
     string string_buf;
     model_proto.SerializeToString(&string_buf);
-    
+
     TensorrtLogger trt_logger(nvinfer1::ILogger::Severity::kWARNING);
     auto trt_builder = unique_pointer<nvinfer1::IBuilder>(nvinfer1::createInferBuilder(trt_logger));
     auto trt_network = unique_pointer<nvinfer1::INetworkDefinition>(trt_builder->createNetwork());
     auto trt_parser = unique_pointer<nvonnxparser::IParser>(nvonnxparser::createParser(*trt_network, trt_logger));
     trt_parser->parse(string_buf.data(), string_buf.size());
-    trt_builder->setMaxBatchSize(kMaxBatchSize);
-    trt_builder->setMaxWorkspaceSize(kMaxWorkSpaceSize);
+
+    const char* batch_env = getenv("TENSORRT_MAX_BATCH_SIZE");
```

Comment:
would be better if the env variable started with ORT_
to ensure we don't potentially collide with real TENSORRT env variables